packages/lab/src/file-drop-zone/FileDropZone.tsx

```diff
@@ -120,193 +41,56 @@ export const FileDropZone = forwardRef<HTMLDivElement, FileDropZoneProps>(
-
+      event.dataTransfer && (event.dataTransfer.dropEffect = "copy");
       if (!isActive && containsFiles(event)) {
         setActive(true);
       }
+      onDragOverProp?.(event);
     };
 
-    const handleDragLeave = () => setActive(false);
-
-    const handleFilesDrop = (event: SyntheticEvent) => {
-      if (!containsFiles(event as DragEvent)) {
-        return setDropResult({
-          event,
-          errors: [INVALID_DROP_TARGET],
-        });
-      }
-
-      const files = extractFiles(event as DragEvent);
-
-      if (files.length > 0) {
-        return setDropResult({
-          event,
-          files,
-          errors: validate ? validateFiles({ files, validate }) : [],
-        });
-      }
+    const handleDragLeave: DragEventHandler<HTMLDivElement> = (event) => {
+      setActive(false);
+      onDragLeaveProp?.(event);
     };
 
     const handleDrop: DragEventHandler<HTMLDivElement> = (event) => {
       event.preventDefault();
-      event.stopPropagation();
-
-      handleFilesDrop(event);
       setActive(false);
+      onDropProp?.(event);
     };
 
-    const handleInputChange: ChangeEventHandler<HTMLInputElement> = (event) => {
-      event.stopPropagation();
-
-      handleFilesDrop(event);
-    };
-
-    // As an ADA requirement when dialog is closed and the focus is returned to the input, we need to
-    // move focus back on the button element so that all labels can be announced correctly
-    const handleInputFocus = useCallback(
-      (event: FocusEvent<HTMLInputElement>) => {
-        event.stopPropagation();
-        buttonRef.current && buttonRef.current.focus();
-      },
-      []
-    );
-
-    const handleInputClick = useCallback(
-      (event: SyntheticEvent<HTMLButtonElement>) => {
-        event.stopPropagation();
-        fileInputRef.current && fileInputRef.current.click();
-      },
-      []
-    );
-
-    const handleKeyDown = useCallback(
-      (event: KeyboardEvent<HTMLButtonElement>) => {
-        if (["Enter", "Space"].indexOf(event.key) !== -1) {
-          fileInputRef.current && fileInputRef.current.click();
-        }
-      },
-      []
-    );
-
-    const fileDropZoneDescription = description ? (
-      <div className="saltFileDropZone-description" id={descriptionId}>
-        {description}
-      </div>
-    ) : null;
-
-    const buttonLabelledBy = (
-      isRejected ? [buttonId, iconId, descriptionId] : [buttonId, descriptionId]
-    ).join(" ");
-
     return (
       <div
-        {...restProps}
         className={clsx(
           withBaseName(),
           {
-            [withBaseName("error")]: isRejected,
+            [withBaseName("error")]: status === "error",
+            [withBaseName("success")]: status === "success",
             [withBaseName("active")]: isActive,
             [withBaseName("disabled")]: disabled,
           },
           className
         )}
-        onDragLeave={handleDragLeave}
+        onDragLeave={!disabled ? handleDragLeave : undefined}
```

Comment:
Can we move this logic to the handleDragLeave function please. It being here makes it harder to follow the logic